client/src/components/Verejne/Map/AddressDetail/ListRow.js

```diff
@@ -23,24 +57,43 @@ const DetailedInfo = compose(
     (state, {id}) => ({
       data: entityDetailSelector(state, id),
     }),
-    {toggleEntityInfo}
+    {toggleEntityInfo, toggleModalOpen, setEntitySearchFor},
   ),
-  withDataProviders(({id}) => [entityDetailProvider(id)])
+  withDataProviders(({id}) => [entityDetailProvider(id)]),
+  withHandlers({
+    openModalSearch: ({toggleModalOpen, setEntitySearchFor, data}) => (e) =>{
+      if (e.stopPropagation) e.stopPropagation()
+      setEntitySearchFor(data.name)
+      toggleModalOpen()
+    }
+  })
```

Comment:
now that this is a fully fledged connected component, we should move it to a separate file (we're keeping one connected component per file, only components like 'Item' which are essentially just helpers can be kept alongside others)